### ISSUE TITLE
BrowserSync + webpack-dev-middleware

### ISSUE THREAD

**Issue by giannico** [Va1/browser-sync-webpack-plugin#6]
Hi, I'm currently running this plugin in Browsersync proxy mode, alongside the webpack-dev-middleware express middleware, it's working really well. One drawback, however, is that I'm running the express server with nodemon, and automatically restarting the express server when I make server-side changes. This causes the Browsersync instance to tear down and start up again, with each server restart (which adds some overhead).

I was attempting to run the Browsersync server outside of the express app (within gulp for instance), and use ths plugin (or the webpack-dev-middleware) to trigger a Browsersync reload, but I can't get a handle to the browsersync instance from within the express app (since it's created/running from gulp). I know this isn't the intent of this plugin, but was wondering if you've experienced anything similar?

Thanks again for the plugin.

**Comment by Va1** [comment]
Hey!

Yes, I once faced the situation similar to what you describe, but I decided that it did not worth the time the fix requires (as it was a small project). 

As I think, there would be no possibility not to restart browsersync on express restart, while browsersync is started in the express app, because nodemon (I guess) is killing the process and starting it again. 

So, if it is worth, basically, what we need is launch browsersync separately, then when express restarts, notify the process in which we did launch browsersync, so it will reload the page. I do not know how to competently do this in node, but have a look at [this project](https://github.com/weixiyen/messenger.js), looks like a cool thing to notify another node process.

Cheers

**Comment by giannico** [comment]
Thanks for the response - glad someone else has faced the same issue!

I ended up working with LiveReload to accomplish this. [tiny-lr](https://github.com/mklabs/tiny-lr), a LiveReload server implementation), offers a REST API for triggering browser updates (I wish BrowserSync had a similar implementation, because I do prefer some of the other BrowserSync features).

I ended up creating a simple plugin for webpack which notifies the LiveReload server (that's running outside of express) of just the files that change (so auto css injection works, without a full page refresh). It's my first plugin, so I'm not 100% sure it's the optimal implementation, but it's working well for me so far:

https://github.com/giannico/live-reload-notify-webpack-plugin

I'll definitely check out messenger.js - thanks for the tips!
